### PR TITLE
Choose new staking contracts

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Choose new staking contracts.

Specification:
- User will be presented with a choice of staking contracts for their Predict agent.
- Upon selecting that choice, the script should unstake from any other staking contract and use the indicated one.
- User will be able to select a different staking program by executing `cd trader; poetry run python ./scripts/choose_staking.py --reset; cd ..`

Pending to implement (in future PRs)
- Read current staking program from registry getOwner.
- Remove dependency on Blockscout.
- Determine implementation address from proxy contract.

### CODE REVIEW

File: run_service.sh [hunk 1]
```diff
@@ -683,25 +648,18 @@ suggested_safe_top_up_default=500000000000000000
 
 export RPC_RETRIES=40
 export RPC_TIMEOUT_SECONDS=120
+
+# export CUSTOM_SERVICE_REGISTRY_ADDRESS="0x9338b5153AE39BB89f50468E608eD9d764B755fD"  # Set up in .env file
+# export CUSTOM_STAKING_ADDRESS="0x43fB32f25dce34EB76c78C7A42C8F40F84BCD237"  # Set up in .env file
+# export CUSTOM_OLAS_ADDRESS="0xcE11e14225575945b8E6Dc0D4F2dD4C570f79d9f"  # Set up in .env file
+# export CUSTOM_SERVICE_REGISTRY_TOKEN_UTILITY_ADDRESS="0xa45E64d13A30a51b91ae0eb182e88a40e9b18eD8"  # Set up in .env file
+# export MECH_CONTRACT_ADDRESS="0x77af31De935740567Cf4fF1986D04B2c964A786a"  # Set up in env file
```
Comment (jmoreira-valory) [inline run_service.sh:656]: @kupermind Please validate the following:
1. The parameters commented correspond to addresses that depend/can be inferred from the staking contract.
2. The remaining addresses not commented should remain constant for all staking contracts, for the time being.
3. Please suggest a more appropriate name for any of these variables from point 1 and 2 above.

Comment (kupermind) [reply run_service.sh:656]: CUSTOM_GNOSIS_SAFE_PROXY_FACTORY_ADDRESS => CUSTOM_GNOSIS_MULTISIG_ADDRESS

Comment (jmoreira-valory) [reply run_service.sh:654]: These commented variables will be removed before merging to main.

File: terminate_on_chain_service.sh [hunk 2]
```diff
@@ -120,16 +130,18 @@ export RPC_TIMEOUT_SECONDS=120
...
 export CUSTOM_CHAIN_ID=$gnosis_chain_id
 export CUSTOM_SERVICE_MANAGER_ADDRESS="0x04b0007b2aFb398015B76e5f22993a1fddF83644"
-export CUSTOM_SERVICE_REGISTRY_ADDRESS="0x9338b5153AE39BB89f50468E608eD9d764B755fD"
-export CUSTOM_STAKING_ADDRESS="0x43fB32f25dce34EB76c78C7A42C8F40F84BCD237"
-export CUSTOM_OLAS_ADDRESS="0xcE11e14225575945b8E6Dc0D4F2dD4C570f79d9f"
-export CUSTOM_SERVICE_REGISTRY_TOKEN_UTILITY_ADDRESS="0xa45E64d13A30a51b91ae0eb182e88a40e9b18eD8"
+# export CUSTOM_SERVICE_REGISTRY_ADDRESS="0x9338b5153AE39BB89f50468E608eD9d764B755fD"
+# export CUSTOM_STAKING_ADDRESS="0x43fB32f25dce34EB76c78C7A42C8F40F84BCD237"
+# export CUSTOM_OLAS_ADDRESS="0xcE11e14225575945b8E6Dc0D4F2dD4C570f79d9f"
+# export CUSTOM_SERVICE_REGISTRY_TOKEN_UTILITY_ADDRESS="0xa45E64d13A30a51b91ae0eb182e88a40e9b18eD8"
```
Comment (0xArdi) [inline terminate_on_chain_service.sh:136]: did you mean this?

Comment (jmoreira-valory) [reply terminate_on_chain_service.sh:136]: yes

**Review by Adamantios — APPROVED** [review]

File: terminate_on_chain_service.sh (hunk 2 above)
Comment (jmoreira-valory) [reply terminate_on_chain_service.sh:136]: Update: This will be polished/removed in an upcoming PR.

**Review by 0xArdi — APPROVED** [review]

**Pull request merged**